cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -1643,6 +1707,51 @@ public int countActiveRuns(@Nullable Integer limit) throws IOException {
     return count.get();
   }
 
+  /**
+   * Count all records in launching state.
+   *
+   * @param programTypes set of program types to filter.
+   * @param limit count at most that many runs, stop if there are more.
+   */
+  public int getLaunchingCount(Set<ProgramType> programTypes, @Nullable Integer limit) throws IOException {
+    AtomicInteger count = new AtomicInteger(0);
+    try (CloseableIterator<RunRecordDetail> iterator = queryProgramRuns(
```

Comment:
This is heavy. Can we use io.cdap.cdap.spi.data.StructuredTable#count? If needed, we can even add a field when we write the record to filter efficiently

Reply:
The io.cdap.cdap.spi.data.StructuredTable#count() doesn't support the predicates used in this case. The run record data is stored as string and it is not possible to apply filters on it.

Reply:
In this case we need to add a field and calculate predicate on writing. The counts are used very often and counting in java code would be too slow

Reply:
Added field for `flowcontrolstatus`